src/whylogs/app/logger.py

```diff
@@ -355,46 +357,51 @@ def log_image(self, image, feature_transforms: Optional[List[Callable]] = None,
         track_image(self._profiles[-1]["full_profile"])
 
     def log_csv(self,
-                filepath_or_buffer: FilePathOrBuffer,
-                segments: Optional[Union[List[SegmentTags], List[str]]] = None,
-                profile_full_dataset: bool = False, **kwargs,):
+                filepath_or_buffer: Union[str, Path, IO[AnyStr]],
+                segment_tags: Optional[Union[List[SegmentTags], List[str]]] = None,
```

Comment:
A segment is made up of multiple sets of segment_tags.

Reply:
A group of segment tags is a segment (not `segments` in plural form).

Either we create a wrapper object for segment tags and call it `Segment`. I don't like having a `list` parameter but calling it singular - it's very confusing from API design perspective.

Reply:
yes, but the user needs to define multiple segments.

Reply:
This is so we can have overlapping segments.

Reply:
A segment = List[SegmentTags]

In order to support multple segments, the API must accept `List[List[SegmentTags]]`. Maybe I'm missing some context with the API here?

Reply:
My bad - misunderstood `SegmentTags` (plural) vs `SegmentTag`. Renamed `SegmentTags` to `Segment` to make it clear that it represents a Segment